tests/cpp_unit_tests/test_distribution.cpp

```diff
@@ -0,0 +1,63 @@
+#include "openmc/distribution.h"
+#include "openmc/random_lcg.h"
+#include <catch2/catch_test_macros.hpp>
+#include <pugixml.hpp>
+
+TEST_CASE("Test alias method sampling of a discrete distribution")
```

Comment:
Not sure what the testing standards are for random stuff in OpenMC, but this has a 1% chance of failing by relying on statistical measures.   I'd recommend creating some carefully designed PDFs and then sampling them with a pre-defined stream of variables (not random) and ensuring that you get the sample you want.

Reply:
Hi @gonuke, thanks for the review! That test mimics `test_discrete` in the `tests/unit_tests/test_stats.py`; however, I do see that test uses 4 standard deviations as opposed to 3 so I will increase it so the chance of failure is smaller. The C++ unit testing with catch2 is something @paulromano wants.